src/lib.rs

```diff
@@ -191,7 +192,7 @@ impl<'a, T> PathTree<'a, T> {
         self
     }
 
-    pub fn find<'b>(&'a self, path: &'b str) -> Option<Path<'a, 'b, T>> {
+    pub fn find<'a, 'b>(&'a self, path: &'b str) -> Option<Path<'a, 'b, T>> {
```

Comment:
I think the lifetime for path would be fine to keep. For PyO3 we need to copy the data into Python objects to return it anyway, so lifetimes on methods are fine. The problematic lifetimes are on the PathTree. Also I imagine most use cases don’t care too much about insertion times (ideally that would happen once at app startup) so I would heavily bias toward optimizing the performance of find() over all else.

Reply:
Is this change better? I decided to make it an elided lifetime instead of explicit.

Reply:
I think so, I’ll try updating the wrapper and confirm. By the way, if you have any input there feel free to open a PR, I’m still learning so I’m sure you could improve it.